### DISCUSSION TITLE
Impersonate User on Frontend?

### DISCUSSION THREAD

**Discussion by osseonews** [nhost/nhost#2751] [Q&A]
We are trying to create an ability for specific admins to be able to log in into our frontend react application to impersonate a user. Our approach (because we need this to happen on the frontend, where we use Nhost client and nhostprovider), is that we have a serverless function, which after checking for admin rights from an admin that is logged in, takes the user we want to impersonate and creates a refresh token. We then send the refreshToken information back to the frontend and store these in localstorage: nhostRefreshTokenId, nhostRefreshToken, and nhostRefreshTokenExpiresAt, overwriting the current tokens in localstorage. Our question is: how do we then generate an accessToken on the frontend for the new refresh tokens? It doesn't seem like the accessToken is stored anywhere on the frontend, so after we update the localstorage with the new refreshToken information of the impersonated user, how do we then initiate a new session and create the new Access Token that then becomes the new accessToken?

Hope this makes sense. We couldn't figure out any other solution for impersonating a user on the frontend of a react application (doing it on the backend is easy, but we need the admin to navigate the website as the user).

Thanks.

**Comment by dbarrosop** [comment]
You should be able to use the following method:

https://docs.nhost.io/reference/javascript/auth/refresh-session

which should handle all the magic for you, no need to modify the localStorage

**Comment by osseonews** [comment]
Thanks. So we would jsut send back the refreshToken for the user from the backend, and then do nhost.auth.refreshToken(refreshToken) and the frontend will handle it all and update everything with the new impersonated user?
